### DISCUSSION TITLE
web.whatsapp.com text field not updated after exiting nvim

### DISCUSSION THREAD

**Discussion by Praful** [glacambre/firenvim#1649] [General]
I can trigger nvim in web.whatsapp.com. `:w` writes the text (I can see it in the temp file). 
However, when I quit using `:q` or `:wq`, the text is not written back.

Is there a way to specify the selector, firenvim should write back to? 

I'm using the latest firefox on Linux Mint. I have tried brave too.

Other sites are fine.

**Comment by glacambre** [comment]
You can configure what selector Firenvim takes over with the [selector](https://github.com/glacambre/firenvim?tab=readme-ov-file#configuring-what-elements-firenvim-should-appear-on) configuration attribute.

However, I'm afraid changing the selector won't help here. I believe whatsapp web is written in a fancy JavaScript framework (react, most likely), which Firenvim can't interact with.

**Reply by Praful** [reply]
I first asked chatgpt this question and it said

```
vim.g.firenvim_config = {
    localSettings = {
        ['web.whatsapp.com'] = {
            takeover = 'always',
            postexit = [[
                document.querySelector('span.selectable-text.copyable-text[data-lexical-text="true"]').innerText = firenvim.get_text();
            ]]
        }
    }
}

```
And added that the selector should be changed. Does `postexit` exist?!

Is it possible that the selector used to pick what text needs to be edited is different from the selector used to write the text back? If so,  can this be done (the chatgpt suggestion didn't work!).
